### PR TITLE
enforce length `n` serial distribution

### LINKED ISSUE #111: `simulate_tree` example does not conform with documentation
Documentation says `serials_dist` should be:

> function with only one argument `n`, representing the number of serial intervals to generate

but the example suggests

```{r}
  serials_dist = function(x) 3,
```

I would suggest that we enforce length `n` here so as to avoid any unexpected behaviour.

### PR DESCRIPTION
**Pull request by sbfnk**
Fixes #111

### CODE REVIEW

**Comment by jamesmbaazam** [comment]
Checking now.

File: R/checks.R [hunk 1]
```diff
@@ -45,6 +45,13 @@ check_serial_valid <- function(serials_dist) {
       "(see details in ?sim_chain_tree)."
     ))
   }
+  x <- serials_dist(10)
+  if (!checkmate::test_numeric(x, len = 10)) {
+    stop(
+      "The return values of `serials_dist` must be a numeric vector of length ",
+      "`n`."
+    )
+  }
```
Comment (jamesmbaazam) [inline R/checks.R:54]: I like this added check. I can also confirm this is causing the changes in the snapshot tests.

**Review by jamesmbaazam — COMMENTED** [review]
Thanks for these changes, Seb. I've left a comment to hopefully help us fix the CI issue.

**Comment by sbfnk** [comment]
> There's something weird going on here that I am yet to diagnose. seeds have been set in each test_that() call but it seems they're being lost after each function call. I'm not getting the same errors in `main` and not sure why it's happening here. Will investigate further.

Ohhh is the issue that in the tests we're generating a new random number (in `check_serial_valid`) and therefore all subsequent calls to random number generator calls different (pseudo-)random numbers than without that call. In that case all I need to do is accept the snapshot, right?

**Comment by jamesmbaazam** [comment]
> Ohhh is the issue that in the tests we're generating a new random number (in `check_serial_valid`) and therefore all subsequent calls to random number generator calls different (pseudo-)random numbers than without that call.

Yes, exactly.

> In that case all I need to do is accept the snapshot, right?

Yep. That should fix it.

**Pull request merged**